service-registry/src/main/java/io/servicecomb/serviceregistry/client/LocalServiceRegistryClientImpl.java

```diff
@@ -267,6 +270,43 @@ protected Microservice findLatest(String appId, String serviceName, VersionRule
     return allInstances;
   }
 
+  @Override
+  public MicroserviceInstances findServiceInstances(String selfMicroserviceId, String appId, String serviceName,
```

Comment:
you copy old findServiceInstance, and do some modify
DO NOT do this things.

can extract a method, all invoke the new one
or one invoke another
BUT DO NOT COPY CODE

Reply:
also local registry must impl revision logic
otherwise will print too many logs in pull mode.

Reply:
done